### PR TITLE
WIP doc: auto-generate API documentation for the machine package

### PR DESCRIPTION
**Pull request by aykevl**
Work-in-progress documentation for the machine package. Sharing it now for others to see, in case there is a better way to do this.

The build tags etc. should be provided by the `tinygo` command. I have some code prepared that can expose these things.

### CODE REVIEW

**Comment by deadprogram** [comment]
What is the status with this branch?

**Comment by aykevl** [comment]
Updated. I'm not sure why it didn't work before, but now it does. It is able to generate documentation for a whole list of microcontrollers.

This is not yet finished, the docs probably need to be moved around a bit. Also, it should somehow be integrated with the "microcontrollers" sections but I'm not quite sure how, maybe as a subpage of every supported board? Any ideas?

This PR depends on the `target` subcommand: https://github.com/tinygo-org/tinygo/pull/692

File: doc-gen/main.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+	for _, target := range os.Args[1:] {
+		fmt.Println("Generating documentation for:", target)
+		path := filepath.Join("..", "content", "machine", target+".md")
+		writeTargetDoc(target, path)
+	}
+}
+
+func writeTargetDoc(target, path string) {
+	// Get the important information from the compiler.
+	cmd := exec.Command("tinygo", "target", target)
```
Comment (deadprogram) [inline doc-gen/main.go]: This needs be updated to `info`, right?

Comment (aykevl) [reply doc-gen/main.go]: Fixed.

**Comment by aykevl** [comment]
The question now really is: where to put the documentation? It would be nice to include the docs on the tinygo.org website with the next release.

**Comment by deadprogram** [comment]
No longer needed, so closing.

**Pull request closed without merging**